src/ipc/ipc-common/Registry.h

```diff
@@ -33,39 +33,61 @@
 #include <QMap>
 #include <functional>
 #include <QVector>
+#include <QDebug>
+
+struct RegistryEntry {
+    using ListenerFunction = std::function<void()>;
+
+    QObject* context = nullptr;
+    ListenerFunction function;
+
+};
 
 template<typename Type, typename Key = QString>
 class Registry {
 public:
-    using ListenerFunction = std::function<void()>;
 
     template<typename SourceObjectType, typename SourceSignalType>
     void addListener(const Key& key, SourceObjectType *receiver, void (SourceSignalType::*receiverMethod)()) const
     {
         auto& listeners = m_listeners[key];
-        QPointer<SourceObjectType> p(receiver);
-        listeners.append([p, receiverMethod]() {
-            if (p) {
-                auto *receiver = p.data();
-                (receiver->*receiverMethod)();
+        RegistryEntry e;
+        e.context = receiver;
+        e.function = [receiverMethod, receiver]() {
+            (receiver->*receiverMethod)();
+        };
+        listeners.append(e);
+        QObject::connect(receiver, &QObject::destroyed, [this, receiver,key] () {
```

Comment:
Potential source of crash if registry is destroyed before connection (unlikely, but not excluded). Please create private slot for this, it will be more elegant than lambda. It will also follow SRP then. 
One more reason to extract this lambda: having method removeListener will improve API.

Reply:
Added "owner" argument.
"key" needs to be captured to avoid iteration over the whole map.